django/core/validators.py

```diff
@@ -144,7 +144,9 @@ class EmailValidator(object):
     domain_regex = re.compile(
         # max length of the domain is 249: 254 (max email length) minus one
         # period, two characters for the TLD, @ sign, & one character before @.
-        r'(?:[A-Z0-9](?:[A-Z0-9-]{0,247}[A-Z0-9])?\.)+(?:[A-Z]{2,6}|[A-Z0-9-]{2,}(?<!-))$',
+        # max length of the gTDs is 63
```

Comment:
```python
r'((?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+)(?:[A-Z0-9-]{2,63}(?<!-))$'
```
This is what I came up with. Each of the label can be as long as 63. But now, total length can be greater than 255.
Any suggestions on how to handle it from regex? Or should I check the length just before returning True or False in line no. 188?